hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -1042,22 +1046,27 @@ public static void aggregateColumnStats(IndexedRecord record, Schema schema,
 
     schema.getFields().forEach(field -> {
       Map<String, Object> columnStats = columnToStats.getOrDefault(field.name(), new HashMap<>());
-      final String fieldVal = getNestedFieldValAsString((GenericRecord) record, field.name(), true, consistentLogicalTimestampEnabled);
+      final Object fieldVal = getNestedFieldVal((GenericRecord) record, field.name(), true, consistentLogicalTimestampEnabled);
+      final Schema fieldSchema = getNestedFieldSchemaFromRecord((GenericRecord) record, field.name());
       // update stats
-      final int fieldSize = fieldVal == null ? 0 : fieldVal.length();
+      final int fieldSize = fieldVal == null ? 0 : StringUtils.objToString(fieldVal).length();
```

Comment:
This is incorrect value-size calculation: string size != binary size

Reply:
I actually don't think this metric is useful at all without fixing the encoding we use for the payload. Is the encoding here Avro? If it's Avro, then the question is ...why Avro?

I'd suggest before we try to implement some of these metrics let's start with use-case we have in mind for them and then take it up from there

Reply:
For ex, Parquet does provide such metrics but these are also much more sensible at Parquet level: encoding is fixed (Thrift), it also directly translates into size on disk. In our case i don't think there's a sensible high-level use-case for such metric

Reply:
This method is used for meging colstats when handing updates for deltacommits. In this case, the records are encoded in avro by design.
I agree with your point about size. While there is no immediate use-case for size, it might be helpful in future for e.g. skewness. Since we have the right size for base files, do you think we can set it to 0 here for now?